workspaces/optic-engine/src/projections/shape.rs

```diff
@@ -416,26 +434,15 @@ impl ShapeProjection {
   }
 
   pub fn without_field(&mut self, field_id: FieldId) {
-    let field_node_index = self
-      .get_field_node_index(&field_id)
-      .expect("expected field_id to have corresponding node");
-
-    let object_edge_index = self
-      .graph
-      .edges_directed(*field_node_index, petgraph::Direction::Outgoing)
-      .find(|parent_edge| matches!(parent_edge.weight(), Edge::IsFieldOf))
-      .map(|object_edge| object_edge.id());
+    let node_index = self
+      .node_id_to_index
+      .get(&field_id)
+      .expect("field to exist in node to index map");
 
-    if let Some(object_edge_index) = object_edge_index {
-      self.graph.remove_edge(object_edge_index); // prevents field to be resolved from the object shape node
+    match self.graph.node_weight_mut(*node_index) {
+      Some(Node::Field(field_node)) => field_node.is_removed = true,
+      _ => {}
     }
-
-    self.node_id_to_index.remove(&field_id); // prevents field node to be looked up by field id
-
-    // GOTCHA: we're not deleting the field node itself, as that would invalidate self.node_id_to_index
-    // as the graph indexes shift.
-    // TODO: figure out the implications of the above, like correctness of queries and
-    // eventual garbage collection.
```

Comment:
I don't think we can get away with this. In a graph representation, the entire point is that you can construct queries by walking it, and rely on edges being representative. Adding the "is removed" filter to a couple accessor methods is really brittle, as often the graph is walked directly (as it is with the new edit shape stuff).